### PR TITLE
Code_quality_improvements for idpmgt module

### PR DESCRIPTION
**Pull request by kasungayan**
- Removing plausible Critical, Major and Minor issues related to Application-mgt Module based on SonarQube results.
- Updating Licence Headers.
- Fixing code-review comments.

### CODE REVIEW

File: components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/IdentityProviderManager.java [hunk 1]
```diff
@@ -1090,8 +1101,8 @@ public void updateIdP(String oldIdPName, IdentityProvider newIdentityProvider,
                 } catch (UserStoreException e) {
                     String msg = "Error occurred while retrieving UserStoreManager for tenant "
                             + tenantDomain;
-                    log.error(msg);
-                    throw new IdentityApplicationManagementException(msg);
+                    log.error(msg,e);
```
Comment (hpmtissera) [inline components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/IdentityProviderManager.java]: code not formatted

Comment (kasungayan) [inline components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/IdentityProviderManager.java]: Resolved with #976

File: components/idp-mgt/org.wso2.carbon.idp.mgt.ui/src/main/java/org/wso2/carbon/idp/mgt/ui/client/IdentityProviderMgtServiceClient.java [hunk 2]
```diff
@@ -222,17 +229,17 @@ public void updateIdP(String oldIdPName, IdentityProvider identityProvider) thro
 
             if (fedAuthConfigs != null && fedAuthConfigs.length > 0) {
                 for (FederatedAuthenticatorConfig config : fedAuthConfigs) {
-                    if (!(config.getDisplayName().equals("facebook")
-                            || config.getDisplayName().equals("openid")
-                            || config.getDisplayName().equals("openidconnect")
-                            || config.getDisplayName().equals("samlsso") || config.getDisplayName()
-                            .equals("passovests")))
+                    if (!(("facebook").equals(config.getDisplayName())
+                            || ("openid").equals(config.getDisplayName())
```
Comment (hasinthaindrajee) [inline components/idp-mgt/org.wso2.carbon.idp.mgt.ui/src/main/java/org/wso2/carbon/idp/mgt/ui/client/IdentityProviderMgtServiceClient.java]: Can we use string constants for these

Comment (kasungayan) [inline components/idp-mgt/org.wso2.carbon.idp.mgt.ui/src/main/java/org/wso2/carbon/idp/mgt/ui/client/IdentityProviderMgtServiceClient.java]: Resolved with #85

File: components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/dao/IdPManagementDAO.java [hunk 3]
```diff
@@ -71,21 +89,21 @@
             while (rs.next()) {
                 IdentityProvider identityProvider = new IdentityProvider();
                 identityProvider.setIdentityProviderName(rs.getString(1));
-                if (rs.getString(2).equals("1")) {
+                if (("1").equals(rs.getString("IS_PRIMARY"))) {
```
Comment (prabathabey) [inline components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/dao/IdPManagementDAO.java]: Might be a good idea to introduce appropriate constants upon magic numbers like "1" in order to improve the maintainability of the code.

File: components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/dao/IdPManagementDAO.java [hunk 4]
```diff
@@ -177,9 +195,9 @@
                 Set<Property> properties = new HashSet<Property>();
                 while (proprs.next()) {
                     Property property = new Property();
-                    property.setName(proprs.getString(1));
-                    property.setValue(proprs.getString(2));
-                    if (proprs.getString(3).equals("1")) {
+                    property.setName(proprs.getString("PROPERTY_KEY"));
+                    property.setValue(proprs.getString("PROPERTY_VALUE"));
+                    if (("1").equals(proprs.getString("IS_SECRET"))) {
```
Comment (prabathabey) [inline components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/dao/IdPManagementDAO.java]: Same goes here as well.

File: components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/dao/IdPManagementDAO.java [hunk 5]
```diff
@@ -409,15 +428,15 @@ private ClaimConfig getLocalIdPDefaultClaimValues(Connection dbConnection, Strin
...
-                localClaim.setClaimUri(rs.getString(1));
+                localClaim.setClaimUri(rs.getString("CLAIM_URI"));
 
                 claimMapping.setLocalClaim(localClaim);
                 claimMapping.setRemoteClaim(remoteClaim);
-                claimMapping.setDefaultValue(rs.getString(2));
+                claimMapping.setDefaultValue(rs.getString("DEFAULT_VALUE"));
 
-                if (rs.getString(3).equals("1")) {
+                if (("1").equals(rs.getString("IS_REQUESTED"))) {
```
Comment (prabathabey) [inline components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/dao/IdPManagementDAO.java]: and here as well.

File: components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/internal/IdpMgtListenerServiceComponent.java [hunk 6]
```diff
@@ -33,21 +33,30 @@
  */
 public class IdpMgtListenerServiceComponent {
 
-    private static List<IdentityProviderMgtLister> listners = new ArrayList<IdentityProviderMgtLister>();
+    private static IdpMgtListenerServiceComponent instance = new IdpMgtListenerServiceComponent();
 
-    public static void setIdentityProviderMgtListerService(
+    private List<IdentityProviderMgtLister> listners = new ArrayList<IdentityProviderMgtLister>();
```
Comment (prabathabey) [inline components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/internal/IdpMgtListenerServiceComponent.java]: Was any there any particular reason why this was made singleton? ServiceComponents by convention are not instantiated in other parts of the code, so this is something we can drop.

File: components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/dao/IdPManagementDAO.java (hunk 3 above)
Comment (kasungayan) [inline components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/dao/IdPManagementDAO.java]: Fixed with #1545

File: components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/dao/IdPManagementDAO.java (hunk 4 above)
Comment (kasungayan) [inline components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/dao/IdPManagementDAO.java]: Fixed with #1545

File: components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/dao/IdPManagementDAO.java (hunk 5 above)
Comment (kasungayan) [inline components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/dao/IdPManagementDAO.java]: Fixed with #1545

File: components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/internal/IdpMgtListenerServiceComponent.java (hunk 6 above)
Comment (kasungayan) [inline components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/internal/IdpMgtListenerServiceComponent.java]: Fixed with b1f76c3b02a110631480d1f8636c32e789baa7b8

**Pull request merged**